content/blog/2025-03-24-datafusion-46.0.0.md

```diff
@@ -0,0 +1,92 @@
+
+### New Statistics/Distribution Framework
+
+This release, thanks mainly to [@Fly-Style](https://github.com/Fly-Style) with contributions from [@ozankabak](https://github.com/ozankabak) and [@berkaysynnada](https://github.com/berkaysynnada), includes the initial pieces of a [**redesigned statistics framework](https://github.com/apache/datafusion/pull/14699).** DataFusion’s optimizer can now represent column data distributions using a new `Distribution` enum, instead of the old precision or range estimations. The supported distribution types currently include **Uniform, Gaussian (normal), Exponential, Bernoulli**, and a **Generic** catch-all.
+
+For example, if a filter expression is applied to a column with a known uniform distribution range, the optimizer can propagate that to estimate result selectivity more accurately. Similarly, comparisons (`=`, `>`, etc.) on columns yield Bernoulli distributions (with true/false probabilities) in this model.
+
+This is a foundational change with many follow-on PRs underway. Even though the immediate user-visible effect is limited (the optimizer didn't magically improve by an order of magnitude overnight), but it lays groundwork for more advanced query planning in the future. Over time, as statistics information encapsulated in `Distribution`s get integrated, DataFusion will be able to make smarter decisions like more aggressive parquet pruning, better join orderings, and so on based on data distribution information. The core framework is now in place and is being hooked up to column and table level statistics.
+
+### Aggregate Monotonicity and Window Ordering
+
+DataFusion 46.0.0 adds a new concept of [set](https://github.com/apache/datafusion/pull/14271#)[-monotonicity](https://github.com/apache/datafusion/blob/5210a2bac32e43dc7bf6e7e6000cdeaf2833c06e/datafusion/expr/src/udaf.rs#L1090) for certain transformations, which helps avoid unnecessary sort operations. In particular, the planner now understands when a **window function introduces new orderings of data**. For example, DataFusion now recognizes that a window-aggregate like `MAX` on a column can have an ordering even if the column itself doesn't have an ordering (for certain window frames). PR [#14271](https://github.com/apache/datafusion/pull/14271) introduced a “set-monotonicity” property for window functions, and a follow-up PR [#14813](https://github.com/apache/datafusion/pull/14813) refined the handling of sort order in window frames. Huge thanks to [@berkaysynnada](https://github.com/berkaysynnada) and [@mertak-synnada](https://github.com/mertak-synnada) for this feature.
```

Comment:
I think it would be easier to understand why this feature is valuable if there is some example query showing what type of query now works / is optimized. Not required just a thought